### PR TITLE
bring test coverage of wiki2 up to 100%

### PR DESCRIPTION
**Pull request by zupo**

### CODE REVIEW

**Comment by zupo** [comment]
Refs #2451.

File: docs/tutorials/wiki2/src/tests/tutorial/security.py [hunk 1]
```diff
@@ -9,10 +9,6 @@
 
 
 class MyAuthenticationPolicy(AuthTktAuthenticationPolicy):
-    def authenticated_userid(self, request):
-        user = request.user
-        if user is not None:
-            return user.id
```
Comment (mmerickel) [inline docs/tutorials/wiki2/src/tests/tutorial/security.py]: Sounds good, the rest of the PR looks great!

**Comment by zupo** [comment]
@mmerickel fixed

**Comment by mmerickel** [comment]
Awesome @zupo, thank you!

**Comment by bertjwregeer** [comment]
Yay :D

**Comment by stevepiercy** [comment]
Did anyone run the tests successfully after running through the alchemy wiki tutorial from the beginning, I assume using the scaffold and not the cookiecutter?

I am able to get 30 out of 32 tests to pass, so I wanted a sanity check before diving in deeper.

**Comment by stevepiercy** [comment]
I ran through tutorial and the tests in this PR on master, and 2 tests fail. Here's the transcript:

```python
$VENV/bin/py.test -q

========================================================================== ERRORS ==========================================================================
______________________________________________________ ERROR collecting tutorial/tests/test_initdb.py ______________________________________________________
ImportError while importing test module '/Users/stevepiercy/projects/__scaffold-master/pyramidtut/tutorial/tutorial/tests/test_initdb.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tutorial/tests/test_initdb.py:1: in <module>
    import mock
E   ImportError: No module named 'mock'
_____________________________________________________ ERROR collecting tutorial/tests/test_security.py _____________________________________________________
ImportError while importing test module '/Users/stevepiercy/projects/__scaffold-master/pyramidtut/tutorial/tutorial/tests/test_security.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tutorial/tests/test_security.py:1: in <module>
    import mock
E   ImportError: No module named 'mock'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
2 error in 0.64 seconds
```

Although this PR is an improvement on #2451, it does not close it.

I'm currently working on the docs for switching from scaffolds to cookiecutter, and this similar import error arose.

On master I changed the imports in those two files to the following:

```python
import unittest
from unittest import mock
```

But I still get an error:

```python
 $VENV/bin/py.test -q
.................F..............
========================================================================= FAILURES =========================================================================
________________________________________________________________ TestInitializeDB.test_run _________________________________________________________________

self = <tutorial.tests.test_initdb.TestInitializeDB testMethod=test_run>, mocked_sys = <MagicMock name='sys' id='4336964216'>
mocked_session = <MagicMock name='get_tm_session' id='4347812776'>

    @mock.patch('tutorial.scripts.initializedb.get_tm_session')
    @mock.patch('tutorial.scripts.initializedb.sys')
    def test_run(self, mocked_sys, mocked_session):
        from ..scripts.initializedb import main
        main(argv=['foo', 'development.ini'])
>       mocked_session.assert_called_once()

tutorial/tests/test_initdb.py:18: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock name='get_tm_session' id='4347812776'>, name = 'assert_called_once'

    def __getattr__(self, name):
        if name in {'_mock_methods', '_mock_unsafe'}:
            raise AttributeError(name)
        elif self._mock_methods is not None:
            if name not in self._mock_methods or name in _all_magics:
                raise AttributeError("Mock object has no attribute %r" % name)
        elif _is_magic(name):
            raise AttributeError(name)
        if not self._mock_unsafe:
            if name.startswith(('assert', 'assret')):
>               raise AttributeError(name)
E               AttributeError: assert_called_once

../../../../.pyenv/versions/3.5.2/lib/python3.5/unittest/mock.py:583: AttributeError
------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------
2016-12-21 01:53:01,889 INFO  [sqlalchemy.engine.base.Engine:1235][MainThread] SELECT CAST('test plain returns' AS VARCHAR(60)) AS anon_1
2016-12-21 01:53:01,889 INFO  [sqlalchemy.engine.base.Engine:1236][MainThread] ()
2016-12-21 01:53:01,890 INFO  [sqlalchemy.engine.base.Engine:1235][MainThread] SELECT CAST('test unicode returns' AS VARCHAR(60)) AS anon_1
2016-12-21 01:53:01,890 INFO  [sqlalchemy.engine.base.Engine:1236][MainThread] ()
2016-12-21 01:53:01,890 INFO  [sqlalchemy.engine.base.Engine:1140][MainThread] PRAGMA table_info("pages")
2016-12-21 01:53:01,890 INFO  [sqlalchemy.engine.base.Engine:1143][MainThread] ()
2016-12-21 01:53:01,890 INFO  [sqlalchemy.engine.base.Engine:1140][MainThread] PRAGMA table_info("users")
2016-12-21 01:53:01,890 INFO  [sqlalchemy.engine.base.Engine:1143][MainThread] ()
2016-12-21 01:53:01,891 INFO  [sqlalchemy.engine.base.Engine:1140][MainThread] 
CREATE TABLE users (
        id INTEGER NOT NULL, 
        name TEXT NOT NULL, 
        role TEXT NOT NULL, 
        password_hash TEXT, 
        CONSTRAINT pk_users PRIMARY KEY (id), 
        CONSTRAINT uq_users_name UNIQUE (name)
)


2016-12-21 01:53:01,891 INFO  [sqlalchemy.engine.base.Engine:1143][MainThread] ()
2016-12-21 01:53:01,894 INFO  [sqlalchemy.engine.base.Engine:719][MainThread] COMMIT
2016-12-21 01:53:01,895 INFO  [sqlalchemy.engine.base.Engine:1140][MainThread] 
CREATE TABLE pages (
        id INTEGER NOT NULL, 
        name TEXT NOT NULL, 
        data TEXT NOT NULL, 
        creator_id INTEGER NOT NULL, 
        CONSTRAINT pk_pages PRIMARY KEY (id), 
        CONSTRAINT uq_pages_name UNIQUE (name), 
        CONSTRAINT fk_pages_creator_id_users FOREIGN KEY(creator_id) REFERENCES users (id)
)


2016-12-21 01:53:01,895 INFO  [sqlalchemy.engine.base.Engine:1143][MainThread] ()
2016-12-21 01:53:01,896 INFO  [sqlalchemy.engine.base.Engine:719][MainThread] COMMIT
1 failed, 31 passed in 8.71 seconds
```

I have no idea how to interpret this error report, other than it failed. Apologies for my ignorance.

**Comment by zupo** [comment]
Hey @stevepiercy what do you mean by "I ran through tutorial"? If you tell me exact steps I can try to reproduce locally and look for a solution.

**Comment by stevepiercy** [comment]
@zupo thanks for looking into this! I did the following steps on master:

* http://docs.pylonsproject.org/projects/pyramid/en/master/tutorials/wiki2/installation.html
* overwrote my generated tutorial files with those from https://github.com/Pylons/pyramid/tree/master/docs/tutorials/wiki2/src/tests
* ran `pip install -e ".[testing]" to install dependencies
* then pickup from my above transcript, starting with `$VENV/bin/py.test -q`

I did not install the mock package (it was not included in instructions or setup.py), so I had to change the imports.

**Comment by zupo** [comment]
I'm planning to look into this, but cannot do it before after Christmas, sorry!

**Pull request merged**